documentation/projects/proposals/rekognition_data/20240510-implementation_plan_machine_generated_tags_frontend.md

```diff
@@ -0,0 +1,196 @@
+
+When this work is completed, the single result page will display
+machine-generated tags separately from the source tags. For machine-generated
+tags, we will also display the tag provider and its accuracy.
+
+## Step-by-step plan
+
+<!--
+List the ordered steps of the plan in the form of imperative-tone issue titles.
+
+The goal of this section is to give a high-level view of the order of implementation any relationships like
+blockages or other dependencies that exist between steps of the plan. Link each step to the step description
+in the following section.
+
+If special deployments are required between steps, explicitly note them here. Additionally, highlight key
+milestones like when a feature flag could be made available in a particular environment.
+-->
+
+1. [Create a wrapper component that displays the tags in two sections: source-provided and machine generated](#create-a-wrapper-vmediatags-component)
+2. [Add a page that describes the machine-generated and source tags.](#add-about-tags-description-page)
+
+## Step details
+
+<!--
+Describe all of the implementation steps listed in the "step-by-step plan" in detail.
+
+For each step description, ensure the heading includes an obvious reference to the step as described in the
+"step-by-step plan" section above.
+-->
+
+### Create a wrapper `VMediaTags` component
+
+This component will handle filtering the tags into two sets: machine-generated
+and source-provided. It will also display the headings and the link to the
+"About tags" page. A new child component, `VCollapsibleTagSection`, will handle
+displaying of the tags in several rows, collapsing them if there are more than
+three rows of tags.
+
+Currently, we deduplicate all tags to only display unique tags. When we display
+the tags by generation type, there will probably be duplicates between the
+source and the generated tags. However, each section individually will still be
+normalized to prevent duplicates.
+
+The generated tags section should split the tags by different providers into
+subsections, and should use the provider of the tag as the section label. For
+instance, if there are tags by both `recognition` and `clarifai`, there will be
+two subsections in the Generated tags section: tags generated by `recognition`
+and tags generated by `clarifai`. In the future, if there are more providers,
+there could be more tag sections. The tags in each subsection will be ordered by
+their accuracy.
+
+There is no need to create a separate component for machine-generated tags, the
+slot in the `VTag` component will enable us to display the additional accuracy
+information.
+
+### Add "About tags" description page
+
+We should add a page to explain the users the difference between where the
+Openverse tags come from, the difference between the source and generated tags,
+and a short description of the machine-generated tag providers. This page should
+be linked from the Generated tags section.
```

Comment:
I agree with adding content to explain tag are generated, but I disagree with adding a 10th page in the content header.

The current header is quite long, and adding an extra link in the current set doesn't provide enough information about what "tags" is. Instead, I would address the content need as proposed [here](https://github.com/WordPress/openverse/issues/3564#issuecomment-1916380597) for #3564.

The page could be very simple: reuse the current grid and page style, and add a fixed TOC on the left to jump between sections. Since the section will be linked from the search page, the information is introduced with more context about what Openverse is about and how relates to sources and content.

Reply:
I wrote "similar to the sensitive content page" several times, and deleted it. I guess I should have left it in.

What I meant here was we should have a page explaining what the machine-generated tags are, but not add it to the header. The sensitive media page is just like that: you can only navigate it by typing the address in the URL bar, or by opening a sensitive media item and clicking on a link from that page. Do you think this would work, @fcoveram ?